lib/SILOptimizer/Mandatory/TFDevicePartition.cpp

```diff
@@ -151,6 +151,27 @@ void GraphFunctionDeviceInfo::handleDevicePlacement(
        SymbolicValue::getString(deviceString, ctx.getAllocator())});
 }
 
+DeviceType GraphFunctionDeviceInfo::chooseDevice(llvm::StringRef opType) const {
```

Comment:
Consider using a `llvm::StringSwitch`?

Reply:
Noted. I'll see the op device availability can be integrated here, to replace the hard-coded op names. If that does not happen and we need to keep extending `chooseDevice()`, I'll use `llvm::StringSwitch` later.